hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -423,6 +423,14 @@ private void saveWorkloadProfileMetadataToInflight(WorkloadProfile profile, Hood
       logger.info("RDD PreppedRecords was persisted at: " + preppedRecords.getStorageLevel());
     }
 
+    // Cache Latest File-System View & Seal
+    preppedRecords.map(rec -> rec.getPartitionPath()).distinct().collect().stream().forEach(partitionPath -> {
```

Comment:
we should probably test and ensure this addtional `distinct` is okay.. moving this listing out of index does seem like the right/cleaner this to do btw.. (a serious concern I had with the old PR)

Reply:
This is introducing another action on the driver and collecting all records right ? Even after the records are persisted, this may introduce additional overhead.

Reply:
@vinothchandar @n3nash : Added the config with default false. We can discuss f2f on how to measure #liststatus calls.

Reply:
sounds good